app/src/main/java/com/moviefiend/torian/moviefiend/MainActivity.java

```diff
@@ -1,13 +1,26 @@
 package com.moviefiend.torian.moviefiend;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
-public class MainActivity extends AppCompatActivity {
+import com.moviefiend.torian.moviefiend.network.NowPlayingResponse;
+
+public class MainActivity extends AppCompatActivity implements MovieListFragment.Listener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        MovieListFragment fragment = (MovieListFragment) getFragmentManager().findFragmentById(R.id.movie_list_fragment);
+        fragment.setListener(this);
```

Comment:
@ifeins assuming that you are owned by a specific activity or that it implements something is not the best practice IMO. The developer guide has an example in which it uses getActivity with `try-catch` to set a listener inside the fragment (and not pass it) which is just less generic and less explicit.

Regarding using it from other places outside of `onCreate` I don't understand what is the problem. (the fragment doesn't assume it has a listener set)

And regarding Activity destruction, isn't the fragment get destructed first anyway?